CMakeLists.txt

```diff
@@ -1,45 +1,127 @@
-cmake_minimum_required(VERSION 2.8.12)
-project(abb_librws)
+cmake_minimum_required(VERSION 3.5)
 
-find_package(catkin REQUIRED COMPONENTS cmake_modules)
+# Read version from the package.xml file.
+file(READ ${CMAKE_CURRENT_SOURCE_DIR}/package.xml package_xml_str)
+if(NOT package_xml_str MATCHES "<version>([0-9]+.[0-9]+.[0-9]+)</version>")
+  message(FATAL_ERROR "Could not parse project version from package.xml. Aborting.")
+endif()
+
+# At this point we either have a proper version string, or we've errored
+# out with a FATAL_ERROR above. So assume CMAKE_MATCH_1 contains our
+# package's version.
+project(abb_librws VERSION ${CMAKE_MATCH_1} LANGUAGES CXX)
+
+include(GNUInstallDirs)
+
+list(INSERT CMAKE_MODULE_PATH 0 "${CMAKE_CURRENT_SOURCE_DIR}/cmake/modules/third_party")
+
+if(WIN32)
+  set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
+endif()
 
 ########################
 ## POCO C++ Libraries ##
 ########################
-# we need at least 1.4.3 because of websocket support
-find_package(Poco 1.4.3 REQUIRED COMPONENTS Net Util Foundation XML)
-
-###################################
-## catkin specific configuration ##
-###################################
-catkin_package(INCLUDE_DIRS include
-               LIBRARIES ${PROJECT_NAME}
-               DEPENDS Poco)
+# We need at least 1.4.3 because of WebSocket support.
+find_package(Poco 1.4.3 REQUIRED COMPONENTS Foundation Net Util XML)
 
 ###########
 ## Build ##
 ###########
-set(SRC_FILES
+if(NOT DEFINED BUILD_SHARED_LIBS)
+  option(BUILD_SHARED_LIBS "Build dynamically-linked binaries" ON)
+endif()
+
+set(
+  SRC_FILES
     src/rws_client.cpp
     src/rws_common.cpp
     src/rws_interface.cpp
     src/rws_poco_client.cpp
     src/rws_rapid.cpp
-    src/rws_state_machine_interface.cpp)
-
-include_directories(include
-                    ${Poco_INCLUDE_DIRS}
-                    ${catkin_INCLUDE_DIRS})
+    src/rws_state_machine_interface.cpp
+)
 
 add_library(${PROJECT_NAME} ${SRC_FILES})
-target_link_libraries(${PROJECT_NAME} ${Poco_LIBRARIES} ${catkin_LIBRARIES})
+add_library(${PROJECT_NAME}::${PROJECT_NAME} ALIAS ${PROJECT_NAME})
+
+include(GenerateExportHeader)
+generate_export_header(${PROJECT_NAME})
+
+target_include_directories(${PROJECT_NAME} PUBLIC
+  "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include;${CMAKE_CURRENT_BINARY_DIR}>"
+  $<INSTALL_INTERFACE:$<INSTALL_PREFIX>/include>
+  ${Poco_INCLUDE_DIRS}
```

Comment:
Should this be scoped?

@traversaro: what do you think?

Reply:
Leaving it as unscoped is ok, as Poco is used in the public headers, and so downstream compilation unit that use `abb_librws`need also to include the poco include directories. This will make the package non-relocatable, but as discussed before that is  expected as we are not consuming Poco using imported targets.